src/dbnode/storage/index.go

```diff
@@ -740,11 +746,26 @@ func (i *nsIndex) Tick(c context.Cancellable, startTime time.Time) (namespaceInd
 	return result, multiErr.FinalError()
 }
 
-func (i *nsIndex) Flush(
+func (i *nsIndex) WarmFlush(
 	flush persist.IndexFlush,
 	shards []databaseShard,
 ) error {
-	flushable, err := i.flushableBlocks(shards)
+	return i.flush(flush, shards, series.WarmWrite)
```

Comment:
hmm I don't think this is what that `series.WarmWrite` constant is supposed to be used for. I'd prefer you just passed a bool or made your own constant than overload this one which is used on the write path and has a very specific meaning (unless we've already established this pattern somewhere which is possible)